src/main/java/com/palantir/refreshable/DefaultRefreshable.java

```diff
@@ -45,20 +47,16 @@
  */
 final class DefaultRefreshable<T> implements SettableRefreshable<T> {
     private static final Logger log = LoggerFactory.getLogger(DefaultRefreshable.class);
-    /**
-     * Every ten times refreshable.map is called without an update we must purge SelfRemovingMapSubscriber instances
-     * whose children have been reaped.
-     */
-    private static final int CLEAN_THRESHOLD = 10;
+    private static final Cleaner REFRESHABLE_CLEANER = Cleaner.create();
 
     private static final int WARN_THRESHOLD = 1000;
 
-    @GuardedBy("this")
-    private final Set<Consumer<? super T>> orderedSubscribers = new LinkedHashSet<>();
+    private final Set<Consumer<? super T>> orderedSubscribers = Collections.synchronizedSet(new LinkedHashSet<>());
```

Comment:
I'm assuming you need the iteration order guarantee, so can't use ConcurrentHashMap.newKeySet()?

Reply:
Can we add a comment that explains why this guarantee is important?

Reply:
I can dig up the change which introduced subscription order, I think services failed to start due to updates in random order based on object default hash codes.